### ISSUE TITLE
Specify a different DDP connection

### ISSUE THREAD

**Issue by rafaelcorreiapoli** [VeliovGroup/Meteor-Files#235]
- *Describe your feature / request*

Specify a different DDP connection, instead of "Meteor"

 - *How you going to use it? Give a usage example(s)*

I'm trying to upload my files throught a different DDP connection, other than Meteor.

```js
import { DDP } from 'meteor/ddp'
const FilesDDP = DDP(url)
```

It would be nice if we could pass FilesDDP as the ddp connection to the constructor, than the methods would be called using this connection

**Comment by dr-dimitru** [comment]
Hi @rafaelcorreiapoli ,

Great suggestion.
 1. Could you point me to the right direction in Meteor docs about how to create and use custom DDP connection?
 2. Is this required per upload or per collection on your end?

**Comment by dr-dimitru** [comment]
@rafaelcorreiapoli 

Any news on your end?
Also, this feature request is partly covered in #51 , is there any chance to fulfil this issue and #51 ?

**Comment by dr-dimitru** [comment]
@rafaelcorreiapoli could you please answer on my questions above?

**Comment by rafaelcorreiapoli** [comment]
Hi @dr-dimitru, sorry for the late reply.
I implemented it like this:
```
export const FileDDP = DDP.connect('http://localhost:4000')

const collection = new Mongo.Collection('files', {
  connection: FileDDP,
})

export default new FilesCollection({
  ddp: FileDDP,
  debug: true,
  collection,
  allowClientCode: false, // Disallow remove files from Client
  onBeforeUpload(file) {
    if (/png|jpg|jpeg|bmp/i.test(file.extension) || file.type === 'image/webp') {
      return true;
    } else {
      return 'Please upload image, with size equal or less than 10MB';
    }
  }
});
```

DDP.connect returns a new instance of a ddp connect ("Meteor" itself is created like this, I think)

Then I changed *files.coffee* to use this ddp instance I passed in the config
```
@config.ddp.call @collection._methodNames._Write, opts, (error) ->
```

Since I don't know nothing about coffeescript, I think I did a naive implementation and it would not be good for a PR

But I think you got the idea! If you could publish this, it would be great!

**Comment by dr-dimitru** [comment]
Implemented in  [`v1.7.4`](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.7.4)
